tests/test_validators.py

```diff
@@ -63,6 +63,7 @@ class Model(BaseModel):
     assert Model(x='1.0').x == 1.0
 
 
+@pytest.mark.xfail(sys.version_info >= (3, 9) and sys.implementation == 'pypy', reason='PyPy 3.9+ bug')
```

Comment:
I attempted to reproduce locally but could not; I suggest to move forward we xfail this one for now, can revisit another time.

Reply:
Sounds good. This one has been funky for me as well.

Reply:
Looks like this test still ran for some reason, I wonder if it was the `sys.implementation` check that didn't work...

Reply:
Ah yes it's `sys.implementation.name`.